### PR TITLE
Add version selector to chart and release editor

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

File: app-catalog/src/components/charts/EditorDialog.tsx [hunk 1]
```diff
@@ -100,12 +107,18 @@ export function EditorDialog(props: {
...
     setIsFormSubmitting(true);
-    if (!validateFormFields()) {
+    if (!validateReleaseNameFormField()) {
       enqueueSnackbar(`Release name is required`, {
         variant: 'error',
       });
       return;
     }
+    if (!validateVersionFormField()) {
+      enqueueSnackbar(`Version is required`, {
```
Comment (joaquimrocha) [inline app-catalog/src/components/charts/EditorDialog.tsx]: Should not be a literal.

File: app-catalog/src/components/releases/EditorDialog.tsx [hunk 2]
```diff
@@ -106,6 +120,13 @@ export function EditorDialog(props: {
       });
       return;
     }
+    if (!selectedVersion) {
+      enqueueSnackbar(`Please select a version`, {
```
Comment (joaquimrocha) [inline app-catalog/src/components/releases/EditorDialog.tsx]: Should not be a literal.

File: app-catalog/src/components/releases/List.tsx [hunk 3]
```diff
@@ -11,10 +11,14 @@ import { useEffect, useState } from 'react';
 import { listReleases } from '../../api/releases';
 
 export default function ReleaseList() {
-  const [releases, setReleases] = useState([]);
+  const [releases, setReleases] = useState(null);
```
Comment (joaquimrocha) [inline app-catalog/src/components/releases/List.tsx]: Should use something like `useState<type>...`

**Review by joaquimrocha — CHANGES_REQUESTED** [review]

File: app-catalog/src/components/releases/List.tsx (hunk 3 above)
Comment (ashu8912) [reply app-catalog/src/components/releases/List.tsx]: updated

File: app-catalog/src/components/releases/EditorDialog.tsx (hunk 2 above)
Comment (ashu8912) [reply app-catalog/src/components/releases/EditorDialog.tsx]: updated

File: app-catalog/src/components/charts/EditorDialog.tsx (hunk 1 above)
Comment (ashu8912) [reply app-catalog/src/components/charts/EditorDialog.tsx]: updated

File: app-catalog/src/components/charts/EditorDialog.tsx [hunk 4]
```diff
@@ -99,8 +106,14 @@ export function EditorDialog(props: {
...
-    if (!validateFormFields()) {
-      enqueueSnackbar(`Release name is required`, {
+    if (!validateReleaseNameFormField()) {
+      enqueueSnackbar("Release name is required", {
+        variant: 'error',
+      });
+      return;
+    }
+    if (!validateVersionFormField()) {
+      enqueueSnackbar("Version is required", {
```
Comment (joaquimrocha) [inline app-catalog/src/components/charts/EditorDialog.tsx:116]: If we always have a version selected, will this ever happen?

**Pull request merged**